node_lambnik/src/tiler/src/tiler.js

```diff
@@ -17,21 +17,42 @@ const TILE_WIDTH = 256
 // Register plugins
 mapnik.register_default_input_plugins()
 
+/**
+ * Disables all layers that aren't specified in the query string
+ */
+const filterLayers = (xmlString, layers) => {
+    // Only filter if there ARE layers to filter on
+    if (layers && layers.length > 0) {
+        return xmlString.replace(
+            /(<Layer name="(.+)")>/g,
+            (fullMatch, beginning, layerName) => {
+                if (!layers.includes(layerName)) {
+                    return `${beginning} status="0">`
+                }
+                return fullMatch
```

Comment:
I think trying to parse markup with regexes is always a dangerous game, and the fact that this very simple attribute addition is not particularly straightforward to read and already seems kind of brittle (can `Layer` have other attributes besides `name`?  If so, this would stop working.) is a decent illustration of why.  I think we should find a (hopefully pretty lightweight, since our needs are quite modest) package to load this as XML and let us traverse it and set attributes as such, then write it back out to a string.

Reply:
I agree, I've realized when writing tests that it already doesn't work even in the simple use cases we've been demoing it on. I'll go swap this out for an XML library, we might need one for YAML too.